test/snapshots/filename-case.mjs.md

```diff
@@ -160,5 +160,26 @@ Generated by [AVA](https://avajs.dev).
 
     `␊
     > 1 | const filename = "foo..JS";␊
-        | ^ File extension \`.JS\` is not in lowercase. Rename it to \`foo..js\`.␊
+        | ^ File extension \`..JS\` is not in lowercase. Rename it to \`foo..js\`.␊
+    `
+
+## invalid(7): const filename = "foo.SPEC.js";
+
+> Input
+
+    `␊
+      1 | const filename = "foo.SPEC.js";␊
+    `
+
+> Filename
+
+    `␊
+    foo.SPEC.js␊
+    `
+
+> Error 1/1
+
+    `␊
+    > 1 | const filename = "foo.SPEC.js";␊
```

Comment:
I'm not sure about this, `.SPEC` not really in `extension`, do we really want it to be lowercased?

Reply:
Sorry, I'm not sure what you mean here - the purpose of this new option is exactly to treat anything after the first dot as an extension instead of the filename.

Reply:
Sorry not been clear.

Before this PR, this rule check two problems, a) the file base name should match required cases, b) the extension should be lowercased.

For file `foo.Test.js` it make sense to enforce `.js` to be lowercased(before), but it doesn't make sense to enforce `.Test` part to be lowercased(this PR).

Reply:
Ok, so you're saying you'd rather not enforce any casing on the parts between the first and last `.`s in the full file name?

Reply:
I think so, unless you have a better idea.

Reply:
I like treating everything after the first dot as part of a multi-part file extension. That's implied by the name of the option (`multipleFileExtensions`). If we'd rather handle it differently, I guess we scrap this PR and start fresh.

Reply:
> I like treating everything after the first dot as part of a multi-part file extension.

It doesn't matter how we treat it, but it doesn't make sense to me to enforce `foo.Test.js` to be `foo.test.js`, at least that's not the purpose of  https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1268

Reply:
Sure, I had no reference to that issue, but if it's not the intent then we probably just want to start fresh here. 👍

Reply:
I figure I can be a bit more constructive here -

I think in the context of a rule that enforces file name casing, if we are enforcing lowercase on the last part of the extension then it makes sense to me that it might be useful to offer the option to enforce lowercase on every part of the extension (or "suffix"). Certainly that's the casing that most of the discussed examples in #1484 seem to follow. Ignoring those extra parts entirely means the rule doesn't help users who would prefer to enforce some kind of convention for them, including myself.

Given this change allows for backwards compatibility, I might suggest we go ahead with the behaviour as-is, and could look to improve the rule further in the future by also adding the ability to ignore those extra parts entirely.